### PR TITLE
Call free on all char allocs

### PR DESCRIPTION
**Pull request by snagles**

### CODE REVIEW

**Review by antalakas — APPROVED** [review]

File: config_iter.go [hunk 1]
```diff
@@ -56,6 +56,9 @@ func (ci *ConfigIter) Free() {
 func (ci *ConfigIter) Here() (*string, *string, error) {
 	var err *C.tiledb_error_t
 	var cparam, cvalue *C.char
+	defer C.free(unsafe.Pointer(cparam))
```
Comment (Shelnutt2) [inline config_iter.go:59]: This is wrong, we don't own these strings its owned by the config iterator in C. We must not free them.

File: context.go [hunk 2]
```diff
@@ -92,9 +91,9 @@ func (c *Context) LastError() error {
 
 	if err != nil {
 		var msg *C.char
+		defer C.free(unsafe.Pointer(msg))
```
Comment (Shelnutt2) [inline context.go:94]: This is wrong, we don't own these strings its owned by the error message in C, we must not free it.

File: query.go [hunk 3]
```diff
@@ -3205,6 +3205,7 @@ func (q *Query) GetFragmentNum() (*uint32, error) {
 // GetFragmentURI returns uri for a fragment
 func (q *Query) GetFragmentURI(num uint64) (*string, error) {
 	var cURI *C.char
+	defer C.free(unsafe.Pointer(cURI))
```
Comment (Shelnutt2) [inline query.go:3208]: We don't own the cURI here, it can't be freeded.

**Review by Shelnutt2 — CHANGES_REQUESTED** [review]
I don't think anything here can be merged.

**Comment by Shelnutt2** [comment]
@snagles please revert this is wrong.

**Pull request merged**